scrapper.py

```diff
@@ -21,43 +36,265 @@ class IncorrectNumberOfArticlesError(Exception):
+
+                pdf_raw_file.download()
+                text = pdf_raw_file.get_text()
+
+                parts_of_article = text.split('Список литературы')
+
+                self.article.text = ''.join(parts_of_article[:-1])
+                break
+
+    def _fill_article_with_meta_information(self, article_bs):
+        """
+        Fills self.article with meta information
+        """
+
+        self.article.title = article_bs.find('h3').get_text()
+
+        tables_bs = article_bs.find_all('table', {"class": "unntable"})
+
+        for table_bs in tables_bs:
+            table_datas_bs = table_bs.find_all('td')
+
+            if table_datas_bs[0].text != 'Авторы':
+                continue
+
+            link_to_author_bs = table_datas_bs[1].find('a')
+
+            self.article.author = link_to_author_bs.text
+
+        text_date = re.search(r'Поступила в редакцию\s+\d{2}\.\d{2}\.\d{4}', self.article.text)
+
+        if text_date:
+            date_re = re.search(r'\d{2}\.\d{2}\.\d{4}', text_date.group(0))
+
+            self.article.date = datetime.strptime(date_re.group(0), '%d.%m.%Y')
+
+
 class Crawler:
     """
     Crawler implementation
     """
+
     def __init__(self, seed_urls, max_articles: int):
-        pass
+        self._seed_urls = seed_urls
+        self.max_articles = max_articles
+        self.urls = []
 
     def _extract_url(self, article_bs):
-        pass
+        """
+        Finds urls from the given article_bs
+        """
+        table_rows_bs = article_bs.find_all('tr', {"class": "unnrow"})
+
+        urls = []
+        overall_urls_count = len(self.urls)
+
+        links_bs = []
+
+        for table_row_bs in table_rows_bs:
+            links_bs.extend(table_row_bs.find_all('a'))
+
+        for link_bs in links_bs:
+            if overall_urls_count + 1 > self.max_articles - PRE_SCRAPPED_URL_COUNT:
+                break
+
+            # Checks if the link leads to an article
+            match = re.match(r'^\?anum', link_bs['href'])
+
+            if not match:
+                continue
+
+            link_url = ''.join([DOMAIN_URL, link_bs['href']])
+
+            if link_url not in pre_scrapped_urls:
+                urls.append(link_url)
+                overall_urls_count += 1
+
+        return urls
 
     def find_articles(self):
         """
         Finds articles
         """
-        pass
+        for seed_url in self._seed_urls:
+            if len(self.urls) + 1 > self.max_articles - PRE_SCRAPPED_URL_COUNT:
+                break
+
+            response = requests.get(seed_url)
+            article_bs = BeautifulSoup(response.text, features="html.parser")
+
+            self.urls.extend(self._extract_url(article_bs))
 
     def get_search_urls(self):
         """
         Returns seed_urls param
         """
-        pass
+        return self._seed_urls
+
+
+class CrawlerRecursive(Crawler):
+    """
+    CrawlerRecursive implementation
+    """
+
+    def find_articles(self):
```

Comment:
I did not see the recursion. Recursive crawler should have *one* seed URL that is *hostname*. Then all the magic is handled inside `CrawlerRecursive`.

Reply:
Done

Reply:
please do not resolve comments yourself.